test/e2e/clusterctl_upgrade.go

```diff
@@ -65,10 +65,12 @@ type ClusterctlUpgradeSpecInput struct {
 	// provider contract to use to initialise the secondary management cluster, e.g. `v1alpha3`
 	InitWithProvidersContract string
 	SkipCleanup               bool
+	PreCreateCluster          func(managementClusterProxy framework.ClusterProxy)
 	PreInit                   func(managementClusterProxy framework.ClusterProxy)
 	PreUpgrade                func(managementClusterProxy framework.ClusterProxy)
 	PostUpgrade               func(managementClusterProxy framework.ClusterProxy)
 	MgmtFlavor                string
+	CNIManifestPath           string
```

Comment:
Is this only required for the initial workload cluster that we create? (that will become a mgmt cluster later)

Reply:
I guess this is needed because metal3 has not implemented installing CNI from ClusterResourceSet and since `CNIManifestPath` still used in https://github.com/kubernetes-sigs/cluster-api/pull/3608 I guess it makes sense to propagate `CNIManifestPath` here for provider still using this input in `ApplyClusterTemplateAndWait`